File: Web Collaboration Tool/Front-end/src/app/Paper/Interfaces/PaperQuestion.ts
import {IQuestionImage} from './IQuestionImage';
import {IPosition} from './IPosition';

export class PaperQuestion {
  public id?: number;

  public text?: string;

  public answer?: string;

  public questionPosition?: IPosition;

  public questionImage?: IQuestionImage;

  public pageNumber?: number;
}

File: Web Collaboration Tool/Back-end/src/main/java/project/webcollaborationtool/Collaboration/Paper/Entities/PaperQuestion.java
package project.webcollaborationtool.Collaboration.Paper.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Data
@Entity
public class PaperQuestion
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Lob
    @NotNull
    private String text;

    @Lob
    @Nullable
    private String answer;

    @Transient
    @JsonInclude
    private Position questionPosition;

    @ManyToOne
    @JsonIgnore
    private Paper examPaper;

    @Nullable
    @OneToOne(cascade = CascadeType.ALL)
    private PaperImage questionImage;
}
